webtau-http/src/test/java/com/twosigma/webtau/http/HttpJavaTest.java

```diff
@@ -0,0 +1,181 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twosigma.webtau.http;
+
+import com.twosigma.webtau.Ddjt;
+import com.twosigma.webtau.http.config.HttpConfiguration;
+import com.twosigma.webtau.http.config.HttpConfigurations;
+import com.twosigma.webtau.utils.CollectionUtils;
+import com.twosigma.webtau.utils.UrlUtils;
+import org.junit.*;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.regex.Pattern;
+
+import static com.twosigma.webtau.Ddjt.*;
+import static com.twosigma.webtau.http.Http.http;
+import static com.twosigma.webtau.utils.CollectionUtils.createMap;
+
+public class HttpJavaTest implements HttpConfiguration  {
+    private static final HttpTestDataServer testServer = new HttpTestDataServer();
+
+    private static final int PORT = 7824;
+
+    @BeforeClass
+    public static void startServer() {
+        testServer.start(PORT);
+    }
+
+    @AfterClass
+    public static void stopServer() {
+        testServer.stop();
+    }
+
+    @Before
+    public void initCfg() {
+        HttpConfigurations.add(this);
+    }
+
+    @After
+    public void cleanCfg() {
+        HttpConfigurations.remove(this);
+    }
+
+    @Test
+    public void simpleObjectMappingExample() {
+        http.get("/end-point-simple-object", (header, body) -> {
+            body.get("k1").should(equal("v1"));
+        });
+    }
+
+    @Test
+    public void simpleListMappingExample() {
+        http.get("/end-point-simple-list", (header, body) -> {
+            body.get(0).get("k1").should(equal("v1"));
+        });
+    }
+
+    @Test
+    public void equalityMatcher() {
+        http.get("/end-point", (header, body) -> {
+            body.get("id").shouldNot(equal(0));
+            body.get("amount").should(equal(30));
+
+            body.get("list").should(equal(Arrays.asList(1, 2, 3)));
+
+            body.get("object").get("k1").should(equal(
+                    Pattern.compile("v\\d"))); // regular expression matching
+
+            body.get("object").should(equal(createMap("k1", "v1", "k3", "v3"))); // matching only specified fields and can be nested multiple times
+
+            body.get("complexList").should(equal(header("k1" , "k2").values( // matching only specified fields, but number of entries must be exact
+                                                        "v1" ,  30,
+                                                        "v11",  40)));
+        });
+
+        http.doc.capture("end-point-object-equality-matchers");
```

Comment:
`HttpGroovyTest` and `HttpTest` seem to be using same doc capture names, won't they override each other?  Or are they being run with different doc roots?

Reply:
if they do it shouldn't matter, idea is tests should be identical in what they assert.